### ISSUE TITLE
process.exit(64)

### ISSUE THREAD

**Issue by hildjj** [tj/commander.js#435]
When you call exit, the return code probably should not be `0` (success).  Traditionally, option parsing errors have resulted in error code 64, as documented in many systems' `sysexits.h` header file, which on my system says:

```
#define EX_USAGE    64  /* command line usage error */
```

I would be happy to submit a PR if you're open to this change.

**Comment by hildjj** [comment]
I'll also fix #188 at the same time.

**Comment by pcdevil** [comment]
@hildjj Hi,
first of all, thanks the pull request, it would be really helpful to use this! (+1 from me)

But I don't think in the current form this fixes #188. It wants an optional exit code for help, mainly because of _something went wrong and we are gracefully exiting_ strategy. Something like this:   
```javascript
function addUser (userName, options) {
	if (!Users.hasUser(userName)) {
		errorOutput('unknown user:', userName);
		program.help(1); // wrong parameters, print help and exit
	}

	// do something awesome
}
```

Of course with your modification you can do this by executing `program.exitCode(1) && program.help();` but it's not _cleaner_ than calling `program.outputHelp() && process.exit(1);`.

It would be nice (but not necessity) to combine these statements.

If you invite me to your reposity I'm happy to collaborate.

**Comment by shadowspawn** [comment]
Referenced from #945 

This issue has not had any activity in over six months. It isn't likely to get acted on due to this report.

Feel free to open a new issue if it comes up again, with new information and renewed interest.
 
Thank you  for your contributions.
